iis-wf/iis-wf-referenceextraction/src/main/java/eu/dnetlib/iis/wf/referenceextraction/patent/OpenPatentWebServiceFacadeFactory.java

```diff
@@ -0,0 +1,90 @@
+    
+    @Override
+    public PatentServiceFacade instantiate(Map<String, String> conf) {
+
+        Preconditions.checkArgument(StringUtils.isNotBlank(conf.get(PARAM_CONSUMER_KEY)));
+        Preconditions.checkArgument(StringUtils.isNotBlank(conf.get(PARAM_CONSUMER_SECRET)));
+
+        Preconditions.checkArgument(StringUtils.isNotBlank(conf.get(PARAM_SERVICE_ENDPOINT_AUTH_HOST)));
+        Preconditions.checkArgument(StringUtils.isNotBlank(conf.get(PARAM_SERVICE_ENDPOINT_AUTH_PORT)));
+        Preconditions.checkArgument(StringUtils.isNotBlank(conf.get(PARAM_SERVICE_ENDPOINT_AUTH_SCHEME)));
+        Preconditions.checkArgument(StringUtils.isNotBlank(conf.get(PARAM_SERVICE_ENDPOINT_AUTH_URI_ROOT)));
+
+        Preconditions.checkArgument(StringUtils.isNotBlank(conf.get(PARAM_SERVICE_ENDPOINT_OPS_HOST)));
+        Preconditions.checkArgument(StringUtils.isNotBlank(conf.get(PARAM_SERVICE_ENDPOINT_OPS_PORT)));
+        Preconditions.checkArgument(StringUtils.isNotBlank(conf.get(PARAM_SERVICE_ENDPOINT_OPS_SCHEME)));
+        Preconditions.checkArgument(StringUtils.isNotBlank(conf.get(PARAM_SERVICE_ENDPOINT_OPS_URI_ROOT)));
+
+        String connectionTimeout = conf.containsKey(PARAM_SERVICE_ENDPOINT_CONNECTION_TIMEOUT)
+                ? conf.get(PARAM_SERVICE_ENDPOINT_CONNECTION_TIMEOUT)
+                : "60000";
+        String readTimeout = conf.containsKey(PARAM_SERVICE_ENDPOINT_READ_TIMEOUT)
+                ? conf.get(PARAM_SERVICE_ENDPOINT_READ_TIMEOUT)
+                : "60000";
+        String throttleSleepTime = conf.containsKey(PARAM_SERVICE_ENDPOINT_THROTTLE_SLEEP_TIME)
+                ? conf.get(PARAM_SERVICE_ENDPOINT_THROTTLE_SLEEP_TIME)
+                : "10000";
+                
+        String retriesCount = conf.containsKey(PARAM_SERVICE_ENDPOINT_RETRIES_COUNT)
+                ? conf.get(PARAM_SERVICE_ENDPOINT_RETRIES_COUNT)
+                : "10";
+                
+        return new OpenPatentWebServiceFacade(Integer.parseInt(connectionTimeout), Integer.parseInt(readTimeout),
+                conf.get(PARAM_SERVICE_ENDPOINT_AUTH_HOST),Integer.parseInt(conf.get(PARAM_SERVICE_ENDPOINT_AUTH_PORT)),
+                conf.get(PARAM_SERVICE_ENDPOINT_AUTH_SCHEME), conf.get(PARAM_SERVICE_ENDPOINT_AUTH_URI_ROOT),
+                conf.get(PARAM_SERVICE_ENDPOINT_OPS_HOST), Integer.parseInt(conf.get(PARAM_SERVICE_ENDPOINT_OPS_PORT)),
+                conf.get(PARAM_SERVICE_ENDPOINT_OPS_SCHEME), conf.get(PARAM_SERVICE_ENDPOINT_OPS_URI_ROOT),
+                buildCredential(conf.get(PARAM_CONSUMER_KEY), conf.get(PARAM_CONSUMER_SECRET)),
+                Long.parseLong(throttleSleepTime), Integer.parseInt(retriesCount));
+    }
+
+    protected static String buildCredential(String key, String secret) {
+        StringBuilder strBuilder = new StringBuilder();
+        strBuilder.append(key);
+        strBuilder.append(':');
+        strBuilder.append(secret);
```

Comment:
This can be replaced with much shorter string concatenation without the need of string builder: `key + ':' + secret`.